pkg/providers/instance/instance.go

```diff
@@ -116,8 +134,23 @@ func (p *DefaultProvider) Create(ctx context.Context, nodeClass *v1.EC2NodeClass
 	if err != nil {
 		return nil, err
 	}
-	efaEnabled := lo.Contains(lo.Keys(nodeClaim.Spec.Resources.Requests), v1.ResourceEFA)
-	return NewInstanceFromFleet(fleetInstance, tags, efaEnabled), nil
+
+	capacityType := p.getCapacityType(nodeClaim, instanceTypes)
```

Comment:
There's a call to getCapacityType() in launchInstance -- can we just do that once at the top of this function